rumqttd/src/router/routing.rs

```diff
@@ -954,12 +952,25 @@ impl Router {
                 cursor,
                 read_count: 0,
                 max_count: 100,
+                // set true for new subscriptions
+                forward_retained_msg,
                 group,
             };
 
             self.scheduler.track(id, request);
             self.scheduler.reschedule(id, ScheduleReason::NewFilter);
             debug_assert!(self.scheduler.check_tracker_duplicates(id).is_none())
+        } else if retain_forward_rule == &protocol::RetainForwardRule::OnEverySubscribe
+            && group.is_none()
```

Comment:
```suggestion
       // If subscription was already present and `RetainForwardRule` is `OnEverySubscribe` set `forward_retained_msg` to true again
        } else if retain_forward_rule == &protocol::RetainForwardRule::OnEverySubscribe
            && group.is_none()
```

nit: its non trivial to know that `HashSet::insert` returns a bool signifying if element was already present can we make it more descriptive by adding a comment there as well